tensorflow/java/src/main/java/org/tensorflow/Shape.java

```diff
@@ -74,7 +74,41 @@ public int numDimensions() {
    * @return The size of the requested dimension or -1 if it is unknown.
    */
   public long size(int i) {
-    return shape[i];
+    return shape == null ? -1 : shape[i];
+  }
+
+  /**
+   * The total number of elements found in a tensor of this shape.
+   * 
+   * <p>If the size of some dimensions is unknown, the total number of elements cannot be calculated and -1 is returned.
+   * 
+   * @return the number of elements or -1 if size of some dimension are unknown
+   */
+  public int numElements() {
+    if (shape == null) {
+       return -1;
+    }
+    int total = 1;
+    for (int i = 0; i < shape.length; ++i) {
+      // TODO (karllessard): There might be a lossy conversion here from 'long' sizes to 'int' total, but this issue
```

Comment:
That's because we use Java nio buffers to transport the data, such as [`FloatBuffer`](https://docs.oracle.com/javase/7/docs/api/java/nio/FloatBuffer.html), and their `allocate` method takes an `int` as parameter. So I guess it is not possible to use such buffers for very large tensors. 

I've been hesitant too, I can convert the `long` to `int` at tensor creation instead, but I left a note there that the issue should be addressed overall the clients because many other places does that lossy conversion (e.g. `Tensor#numElements`). I'll move that note elsewhere I guess?

Reply:
Yeah, let's push the loss to the last point where it is required. So `Shape.numElements()` might as well return a lossless `long`?